service/internal/pipelines/pipelines_test.go

```diff
@@ -0,0 +1,292 @@
+
+				// Validate logs.
+				logsProcessor := pipelines.pipelines[config.NewComponentID(config.LogsDataType)].Processors[i]
+				assert.True(t, logsProcessor.(*testcomponents.ExampleProcessor).Stopped)
+			}
+
+			// Now verify that exporters received data, and are shutdown.
+			for _, expID := range test.exporterIDs {
+				// Validate traces.
+				traceExporter := pipelines.GetExporters()[config.TracesDataType][expID].(*testcomponents.ExampleExporter)
+				require.Len(t, traceExporter.Traces, test.expectedRequests)
+				assert.EqualValues(t, testdata.GenerateTracesOneSpan(), traceExporter.Traces[0])
+				assert.True(t, traceExporter.Stopped)
+
+				// Validate metrics.
+				metricsExporter := pipelines.GetExporters()[config.MetricsDataType][expID].(*testcomponents.ExampleExporter)
+				require.Len(t, metricsExporter.Metrics, test.expectedRequests)
+				assert.EqualValues(t, testdata.GenerateMetricsOneMetric(), metricsExporter.Metrics[0])
+				assert.True(t, metricsExporter.Stopped)
+
+				// Validate logs.
+				logsExporter := pipelines.GetExporters()[config.LogsDataType][expID].(*testcomponents.ExampleExporter)
+				require.Len(t, logsExporter.Logs, test.expectedRequests)
+				assert.EqualValues(t, testdata.GenerateLogsOneLogRecord(), logsExporter.Logs[0])
+				assert.True(t, logsExporter.Stopped)
+			}
+		})
+	}
+}
+
+func TestBuildExportersNotSupportedDataType(t *testing.T) {
+	nopReceiverFactory := componenttest.NewNopReceiverFactory()
+	nopProcessorFactory := componenttest.NewNopProcessorFactory()
+	nopExporterFactory := componenttest.NewNopExporterFactory()
+	badReceiverFactory := newBadReceiverFactory()
+	badProcessorFactory := newBadProcessorFactory()
+	badExporterFactory := newBadExporterFactory()
+
+	tests := []struct {
+		configFile string
+	}{
+		{configFile: "not_supported_exporter_logs.yaml"},
+		{configFile: "not_supported_exporter_metrics.yaml"},
+		{configFile: "not_supported_exporter_traces.yaml"},
+		{configFile: "not_supported_processor_logs.yaml"},
+		{configFile: "not_supported_processor_metrics.yaml"},
+		{configFile: "not_supported_processor_traces.yaml"},
+		{configFile: "not_supported_receiver_traces.yaml"},
+		{configFile: "not_supported_receiver_metrics.yaml"},
+		{configFile: "not_supported_receiver_traces.yaml"},
+		{configFile: "unknown_exporter_config.yaml"},
+		{configFile: "unknown_exporter_factory.yaml"},
+		{configFile: "unknown_processor_config.yaml"},
+		{configFile: "unknown_processor_factory.yaml"},
+		{configFile: "unknown_receiver_config.yaml"},
+		{configFile: "unknown_receiver_factory.yaml"},
+	}
+
+	for _, test := range tests {
+		t.Run(test.configFile, func(t *testing.T) {
+			if test.configFile == "unknown_receiver_config.yaml" {
+				t.Skip("This is a small issue with current implementation which will be fixed in #5512." +
+					"In real binary this will not be hit, since the validation of the config will catch this issue in advance.")
```

Comment:
What's the issue here?

Reply:
The problem is that we build the receivers by looking at configs, and if used in one pipeline we create/initialize it. With this logic if there is a receiver in a pipeline without a config we don't create/initialize it, but this is not a "Valid" config and fails `config.Validate()`.